FWCore/Services/test/SignallingAnalyzer.cc

```diff
@@ -0,0 +1,54 @@
+// -*- C++ -*-
+//
+// Package:     FWCore/Services/test
+// Class  :     SignallingAnalyzer
+//
+// Implementation:
+//     [Notes on implementation]
+//
+// Original Author:  Chris Jones
+//         Created:  Sat, 22 Mar 2014 23:07:05 GMT
+//
+
+// system include files
+
+// user include files
+#include "FWCore/Framework/interface/global/EDAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/allowedValues.h"
+#include <csignal>
+
+class SignallingAnalyzer : public edm::global::EDAnalyzer<> {
+public:
+  SignallingAnalyzer(edm::ParameterSet const& iPSet) : m_signal(iPSet.getUntrackedParameter<std::string>("signal")) {}
+
+  void analyze(edm::StreamID, edm::Event const&, edm::EventSetup const&) const final {
+    if (m_signal == "INT") {
+      raise(SIGINT);
+    }
+    if (m_signal == "ABRT") {
+      raise(SIGABRT);
+    }
+    if (m_signal == "SEGV") {
+      raise(SIGSEGV);
+    }
+    if (m_signal == "TERM") {
+      raise(SIGTERM);
+    }
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& iConf) {
+    edm::ParameterSetDescription desc;
+    desc.ifValue(edm::ParameterDescription<std::string>("signal", "INT", false),
+                 edm::allowedValues<std::string>("INT", "ABRT", "SEGV", "TERM"))
+        ->setComment("which signal to raise.");
+  }
```

Comment:
This is missing e.g.
```suggestion
    desc.ifValue(edm::ParameterDescription<std::string>("signal", "INT", false),
                 edm::allowedValues<std::string>("INT", "ABRT", "SEGV", "TERM"))
        ->setComment("which signal to raise.");
      iConf.addDefault(desc);
  }
```
Is there also a mismatch in the trackedness between the `getUntrackedParameter()` and the `ParameterSetDescription` above?

Reply:
I've run the test and it works. If there was a mismatch it would have failed. I'm pretty sure the 3rd argument of `ParameterSetDescription` is _is this a tracked parameter?_ .

Reply:
The `iConf.addDefault(desc)` (or similar) is missing, so the PSet does not get validated.

Reply:
I added the missing `addDefault` and the test passes :). Thanks for the catch.